api/envoy/type/matcher/v3/regex.proto

```diff
@@ -55,6 +55,15 @@ message RegexMatcher {
 
   // The regex match string. The string must be supported by the configured engine.
   string regex = 2 [(validate.rules).string = {min_len: 1}];
+
+  // Intended for use when matching URL paths. A URL path which contains
+  // reserved characters like ``/abc:+def/`` will be percent encoded as
+  // ``/abc%3A%2Bdef/``.
+  //
+  // When enabled, `percent_decode_input` allows matching the path using the
+  // regex: ``/abc:\\+def/``. When disabled, the regex would instead have to be
+  // written as ``/abc%3A%2Bdef/``.
+  bool percent_decode_input = 3;
```

Comment:
HCM uses `normalize_path`, which makes me wonder if what we want is only to percent decode or to apply the fully path normalization rules?

Reply:
I agree that normalization is super helpful too in this context. As far as I can tell that already works by setting `normalize_path: true`: the path `/../a/%3a` gets rewritten to `/a/%3a` before it reaches this function.

So that's probably good enough (?)